pkg/resourcemanager/controller/secret/reconciler.go

```diff
@@ -66,6 +68,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			// check if we are responsible for this MR, class might have changed, then we need to remove our finalizer
 			if ref.Name == secret.Name && r.ClassFilter.Responsible(&resource) {
 				secretIsReferenced = true
+				if _, ok := secret.Labels[v1alpha1.ReferencedBy]; !ok {
```

Comment:
It would be better if we don't the the patch here. We should rather find here whether Secret is referenced and which MR is referencing it (to later get its name). And after these 2 for loops, we can add our logic for the label patch.